docs/contributing/code/bc.rst

```diff
@@ -0,0 +1,70 @@
+Backward Compatibility Promise
+==============================
+
+Sylius follows a versioning strategy called `Semantic Versioning`_. It means that
+only major releases include BC breaks, whereas minor releases include new features
+without breaking backwards compatibility.
+
+Since Sylius is based on Symfony, our BC promise extends `Symfony's Backward Compatibility Promise`_
+with a few new rules and exceptions stated in this document.
+
+Minor and patch releases
+------------------------
+
+Patch releases (such as 1.0.x, 1.1.x, etc.) do not require any additional work
+apart from cleaning the Symfony cache.
+
+Minor releases (such as 1.1.0, 1.2.0, etc.) require to run database migrations.
+
+Code covered
+------------
+
+This BC promise applies to all of Sylius' PHP code except for:
```

Comment:
Maybe `js` files should be added to this list?

Reply:
JS files aren't PHP code 🎉